compiler/rustc_const_eval/src/interpret/terminator.rs

```diff
@@ -372,14 +372,8 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
         };
         if let (Some(caller), Some(callee)) = (pointee_ty(caller.ty)?, pointee_ty(callee.ty)?) {
             // This is okay if they have the same metadata type.
-            let meta_ty = |ty: Ty<'tcx>| {
-                let (meta, only_if_sized) = ty.ptr_metadata_ty(*self.tcx, |ty| ty);
-                assert!(
-                    !only_if_sized,
-                    "there should be no more 'maybe has that metadata' types during interpretation"
-                );
-                meta
-            };
+            let normalize = |ty: Ty<'tcx>| self.tcx.normalize_erasing_regions(self.param_env, ty);
```

Comment:
```suggestion
            // Even if `ty` is normalized, the search for the unsized tail will project to fields
            // which can yield non-normalized types. So we need to provide a normalization
            // function.
            let normalize = |ty: Ty<'tcx>| self.tcx.normalize_erasing_regions(self.param_env, ty);
```

Reply:
I will extract the normalization change to another PR.